### DISCUSSION TITLE
Kotlin language server issue

### DISCUSSION THREAD

**Discussion by iscsi** [LazyVim/LazyVim#5075] [Q&A]
Hi

I am trying to set up the lazyvim for kotlin development (on raspberry pi 4B):

my neovim version:  NVIM v0.10.2

I've tried to use this command to set up kotlin with LazyVim:

https://www.lazyvim.org/extras/lang/kotlin#:~:text=import%20%3D%20%22-,lazyvim,-.plugins.extras.lang

After this the Mason automatically installed the kotlin-language-server and ktlint.

I wanted to try out a very simple hello world code (so no project files)

I think the kotlin-language-server is not working properly, based on the LspInfo 
![Screenshot 2024-12-14 6 11 55 PM](https://github.com/user-attachments/assets/f8799659-5ad6-4c5f-88d4-053da9a9b461)

the cmd path looks not correct it points to the cwd, but the kotlin-language-server is under the .local/share/nvim/mason/...

another issue, every time I've made a change the ktlint runs and a pop up windows shows a warning: "Formatter 'ktlint' timeout". I guess it is maybe just too slow, how can I increase the time limit ?

Thanks

**Comment by dpetka2001** [comment]
From [nvim-lspconfig docs](https://github.com/neovim/nvim-lspconfig/blob/41e249e7787ab2ca16160bbc4bb4cfca28542995/lua/lspconfig/configs/kotlin_language_server.lua#L10-L17), it looks like the Kotlin LSP server expects any of these files in your root project directory to attach. I don't do Kotlin myself, but this is what the docs say.

**Comment by mikemcgowan** [comment]
I had the exact same problem. I wanted to use Kotlin with Lazyvim. I installed `kotlin-language-server` with Mason via a "lazy extra". 

I've managed to get `kotlin-language-server` working by:

- Installing Gradle 8.12 globally (with SDKMAN!)
- Creating a minimal `build.gradle.kts` in my project's root directory as shown below (thanks to @dpetka2001 for the heads up!)
- Running `gradle build`

```
plugins {
    kotlin("jvm") version "2.0.20"
}

dependencies {
    implementation(kotlin("stdlib", "2.0.20"))
}

repositories {
    mavenCentral()
}
```
